tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -142,77 +144,114 @@ static std::string removeExt(const std::string &FileName) {
+  if (EC)
+    error(Prefix + ": " + EC.message());
+}
+
+static void saveListToFile(std::string Filename, std::string Content) {
+  std::error_code EC;
+  raw_fd_ostream OS{Filename, EC, sys::fs::OpenFlags::OF_None};
+  error(EC, "error opening the file '" + Filename + "'");
+  OS.write(Content.data(), Content.size());
+}
 
-  std::unique_ptr<MemoryBuffer> MB =
-      ExitOnErr(errorOrToExpected(MemoryBuffer::getFileOrSTDIN(InputFile)));
-  std::unique_ptr<Module> M =
-      ExitOnErr(getOwningLazyBitcodeModule(std::move(MB), Context,
-                                           /*ShouldLazyLoadMetadata=*/true));
-  ExitOnErr(M->materializeAll());
+static void saveLLVMModuleToFile(std::string Filename, Module *M) {
+  std::error_code EC;
+  raw_fd_ostream OS{Filename, EC, sys::fs::OpenFlags::OF_None};
+  error(EC, "error opening the file '" + Filename + "'");
+  WriteBitcodeToFile(*M, OS);
+}
 
-  if (OutputFile.empty()) {
+static std::string getResultFilename(std::string InputFile, std::string Ext) {
+  bool MultipleFilesMode = InputFilenames.size() > 1;
+  if (OutputFile.empty() || MultipleFilesMode) {
     if (InputFile == "-")
-      OutputFile = "-";
-    else
-      OutputFile =
-          removeExt(InputFile) +
-          (SPIRV::SPIRVUseTextFormat ? kExt::SpirvText : kExt::SpirvBinary);
+      return "-";
+    return removeExt(InputFile) + Ext;
   }
+  return OutputFile;
+}
 
-  std::string Err;
-  bool Success = false;
-  if (OutputFile != "-") {
-    std::ofstream OutFile(OutputFile, std::ios::binary);
-    Success = writeSpirv(M.get(), Opts, OutFile, Err);
-  } else {
-    Success = writeSpirv(M.get(), Opts, std::cout, Err);
-  }
+static int convertLLVMToSPIRV(const SPIRV::TranslatorOpts &Opts) {
+  bool FileListMode = InputFilenames.size() > 1 && !OutputFile.empty();
+  std::string FileList;
+  for (auto &InputFile : InputFilenames) {
+    bool Success = false;
+    std::string Err;
+    LLVMContext Context;
+    std::unique_ptr<MemoryBuffer> MB =
+        ExitOnErr(errorOrToExpected(MemoryBuffer::getFileOrSTDIN(InputFile)));
+    std::unique_ptr<Module> M =
+        ExitOnErr(getOwningLazyBitcodeModule(std::move(MB), Context,
+                                             /*ShouldLazyLoadMetadata=*/true));
+    ExitOnErr(M->materializeAll());
+
+    std::string ResultFileName = getResultFilename(
+        InputFile,
+        SPIRV::SPIRVUseTextFormat ? kExt::SpirvText : kExt::SpirvBinary);
+
+    if (FileListMode) {
+      FileList = (Twine(FileList) + Twine(ResultFileName) + Twine("\n")).str();
+    }
 
-  if (!Success) {
-    errs() << "Fails to save LLVM as SPIR-V: " << Err << '\n';
-    return -1;
+    if (ResultFileName == "-")
+      Success = writeSpirv(M.get(), Opts, std::cout, Err);
+    else {
+      std::ofstream OFS(ResultFileName.c_str(), std::ios::binary);
+      Success = writeSpirv(M.get(), Opts, OFS, Err);
+    }
+
+    if (!Success) {
+      errs() << "Fails to save LLVM as SPIR-V: " << Err << '\n';
```

Comment:
Why not `error(...)`?

Reply:
I didn't change old style of error reporting, and it's presented in parts of code which I didn't change. If you think that using `error(...)` seems better, I can change it in all places.